core-bundle/src/Resources/contao/dca/tl_image_size.php

```diff
@@ -378,24 +378,53 @@ public function editHeader($row, $href, $label, $title, $icon, $attributes)
 	 */
 	public function getFormats()
 	{
-		$options = array();
-
-		if (\in_array('webp', StringUtil::trimsplit(',', \Config::get('validImageTypes'))))
+		if (!\in_array('webp', StringUtil::trimsplit(',', \Config::get('validImageTypes'))))
 		{
-			$options[] = 'png:webp,png';
-			$options[] = 'jpg:webp,jpg;jpeg:webp,jpeg';
-			$options[] = 'gif:webp,gif';
+			return array();
 		}
 
-		if (!$options)
+		if (!self::isWebpSupported())
```

Comment:
Why did you make this a public static method? We are only using the code here, aren't we?

Reply:
The `getFormats` options callback is likely to be customized and the `isWebpSupported()` check is probably also needed then.